ArduPlane/VTOL_Assist.cpp

```diff
@@ -140,4 +140,94 @@ bool VTOL_Assist::should_assist(float aspeed, bool have_airspeed)
     return force_assist || speed_assist || alt_error.is_active() || angle_error.is_active();
 }
 
+/*
+  check if we are in VTOL recovery
+*/
+bool VTOL_Assist::check_VTOL_recovery(void)
+{
+    const bool allow_fw_recovery =
+        !option_is_set(OPTION::FW_FORCE_DISABLED) &&
+        !quadplane.tailsitter.enabled() &&
+        plane.control_mode != &plane.mode_qacro;
+    if (!allow_fw_recovery) {
+        quadplane.force_fw_control_recovery = false;
+        quadplane.in_spin_recovery = false;
+        return false;
+    }
+
+    // see if the attitude is outside twice the Q_ANGLE_MAX
+    const auto &ahrs = plane.ahrs;
+    const int16_t angle_max_cd = quadplane.aparm.angle_max;
+    if ((abs(ahrs.roll_sensor) > 2*angle_max_cd ||
+         abs(ahrs.pitch_sensor) > 2*angle_max_cd)) {
```

Comment:
May want to consider a circular limit. That is how Q angle max works. 

```(Vector2f{AB_diff[1], -AB_diff[0]}).length() > 2*angle_max_cd```

Will only make a difference in cases with high error in both roll and pitch of course.

Reply:
changed to circular limit, thanks